api-reference/user-app/vitals/vitals.yaml

```diff
@@ -0,0 +1,203 @@
+                type: object
+                properties:
+                  success:
+                    type: boolean
+                    example: true
+                  message:
+                    type: string
+                    example: Vitals updated successfully
+        '400':
+          description: Bad Request - Invalid data or missing parameters
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+        '401':
+          description: Unauthorized - Invalid authentication token
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+        '500':
+          description: Internal Server Error
+      security:
+        - auth: []
+
+components:
+  schemas:
+    Vital:
+      type: object
+      properties:
+        vital_display_name:
+          type: string
+          description: Type of the vital sign (e.g., heart_rate, blood_pressure).
+        vital_loinc_code:
+          type: string
+          description: LOINC code corresponding to the vital sign.
+        vital_value:
+          type: object
+          description: Measured value of the vital sign. Only one type of vital_value is expected (vital_numeric_value or vital_boolean_value or vital_string_value)
+          properties:
+            vital_numeric_value:
+              type: number
+              description: Numeric value of the vital sign.
+            vital_boolean_value:
+              type: boolean
+              description: Boolean value of the vital sign, if applicable.
+            vital_string_value:
+              type: string
+              description: String value of the vital sign, if applicable.
+        unit_display_name:
+          type: string
+          description: Unit of the measured value (e.g., mmHg, mg/dL).
+        unit_ucum_code:
+          type: string
+          description: Unit code as per standard (e.g., {Counts}/min, mg/dL).
+        measured_at:
+          type: object
+          properties:
+            effectiveDateTime:
+              type: string
+              format: date-time
+              description: Date and time when the vital was measured.
+            instant:
+              type: string
+              description: Instant timestamp when the vital was measured.
+            period:
+              type: object
+              properties:
+                startDateTime:
+                  type: string
+                  format: date-time
+                  description: Start date and time of the measurement period.
+                endDateTime:
+                  type: string
+                  format: date-time
+                  description: End date and time of the measurement period.
+          description: Details on when the vital was measured, only one type of value is expected
+
+    ErrorResponse:
+      type: object
+      properties:
+        error:
```

Comment:
this needs to be changed, across Eka we ll be following similar error formats, refer to self-assessment for that

Reply:
copy from there